keps/sig-network/3241-srv-service-name/README.md

```diff
@@ -0,0 +1,987 @@
+_kerberos._udp.<domain> IN SRV <priority> <weight> <port> <target>
+```
+
+The Kubernetes Service object schema currently does not support the
+creation of service objects that have `ports` with the same name but
+a different protocol.  As a consequence, Kubernetes cannot produce
+the SRV records required by such applications.
+
+<!--
+Endpoints and the
+-->
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Extend the ServicePort and EndpointPort types to allow the
+  creation of Service objects that result in SRV records having the
+  same service label but different protocols.
+
+- Alter the behaviour of the Endpoint/EndpointSlice controller to
+  recognise the new schema and propagate the required information to
+  the subordinate Endpoints/EntpointSlice object.
+
+- Coordinate updates to [kubernetes/dns][kubedns], other
+  implementors of the [DNS-based Service Discovery
+  specification][dns-spec] and the ExternalDNS system to recognise
+  the new field and create the correct SRV records.
+
+- If necessary, extend the [DNS-based Service Discovery
+  specification][dns-spec] and other relevant documents to elaborate
+  the requirements with respect to this KEP.
+
+- Maintain backwards compatibility with no change in observable
+  behaviour for Service objects that do not use the extended
+  schema/semantics introduced by this KEP.
+
+[kubedns]: https://github.com/kubernetes/dns
+[dns-spec]: https://github.com/kubernetes/dns/blob/master/docs/specification.md
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Support for URI records or more advanced applications or SRV
+  records such as DNS-SD ([RFC 6763][])
+
+[RFC 6763]: https://www.rfc-editor.org/rfc/rfc6763.html
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a Kerberos administrator, I want to be able to deploy a Kerberos
+KDC in Kubernetes and have Kubernetes create the required SRV
+records automatically (either in KubeDNS and/or via ExternalDNS).
+
+Specifically, the Service object:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: idm
+spec:
+  type: LoadBalancer
+  selector:
+    app: service-test
+  ports:
+  - name: kerberos-udp
```

Comment:
```suggestion
  - name: kerberos-tcp
```
is a typo, right?

Reply:
Yes, thanks.  I shall fix it in the next push.